### PR TITLE
refactor(components): [breadcrumb] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by buqiyuan**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/breadcrumb/src/breadcrumb.vue [hunk 1]
```diff
@@ -9,34 +9,28 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent, provide, ref, onMounted } from 'vue'
+<script lang="ts" setup>
+import { provide, ref, onMounted } from 'vue'
 import { elBreadcrumbKey } from '@element-plus/tokens'
```
Comment (sxzz) [inline packages/components/breadcrumb/src/breadcrumb.vue:14]:
```suggestion
import { breadcrumbKey } from '@element-plus/tokens'
```
Please rename token name

**Review by sxzz — APPROVED** [review]

File: packages/components/breadcrumb/src/breadcrumb-item.vue [hunk 2]
```diff
@@ -12,54 +12,36 @@
...
-    const parent = inject(elBreadcrumbKey, undefined)
-    const ns = useNamespace('breadcrumb')
+const instance = getCurrentInstance()!
+const router = instance.appContext.config.globalProperties.$router as Router
+const parent = inject(breadcrumbKey, undefined)
+const ns = useNamespace('breadcrumb')
 
-    const link = ref<HTMLSpanElement>()
+const separator = parent?.separator
+const separatorIcon = parent?.separatorIcon
```
Comment (sxzz) [inline packages/components/breadcrumb/src/breadcrumb-item.vue:36]:
```suggestion
const { separator, separatorIcon } = parent ?? {}
```

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
Don't forget to add component path to the type safe list in future PR.

https://github.com/element-plus/element-plus/pull/6073/files#diff-a5413da73987503ccadf36f0c1d7d15e945296fa56e968c20d8e09212b2cc366

**Comment by buqiyuan** [comment]
> Don't forget to add component path to the type safe list in future PR.
> 
> https://github.com/element-plus/element-plus/pull/6073/files#diff-a5413da73987503ccadf36f0c1d7d15e945296fa56e968c20d8e09212b2cc366

Ok, I didn't know that existed

**Pull request merged**